### ISSUE TITLE
KeyError -> Method not found

### ISSUE THREAD

**Issue by HasmedFara** [bcb/jsonrpcserver#72]
Hey guys,

When a KeyError exception is raised this is the response {"code": -32601, "message": "Method not found"} which is confused and misleading.

**Comment by bcb** [comment]
The trouble is [here](https://github.com/bcb/jsonrpcserver/blob/4.0.0/jsonrpcserver/dispatcher.py#L145), it's going ahead and assuming the method exists, and then [handling the KeyError](https://github.com/bcb/jsonrpcserver/blob/4.0.0/jsonrpcserver/dispatcher.py#L115) if it doesn't. But KeyError is too generic an exception.

We need something like:
```python
if request.method not in methods.items:
    raise MethodNotFoundError()
```

or a more Pythonic way

```python
try:
    method = methods.items[request.method]
except KeyError:
    raise MethodNotFoundError()
```

**Comment by okke-formsma** [comment]
A similar problem exists for handling of the 'TypeError'. If any code inside a method returns a TypeError, a `InvalidParamsResponse` is thrown instead of a generic `ExceptionResponse`.

**Comment by bcb** [comment]
Unfortunately fixing this would be a breaking change requiring another major release which i'm not comfortable with. Prefer to wait until there are more breaking changes to warrant a version 5.x.

**Comment by bcb** [comment]
Fixed in #95
